packages/wallet-sdk/src/CoinbaseWalletSDK.ts

```diff
@@ -83,28 +65,32 @@ export class CoinbaseWalletSDK {
     return walletLogo(type, width);
   }
 
-  private get walletExtension(): ProviderInterface | undefined {
-    return window.coinbaseWalletExtension;
+  private storeLatestVersion() {
+    const versionStorage = new ScopedLocalStorage('CBWSDK');
+    versionStorage.setItem('VERSION', LIB_VERSION);
   }
+}
 
-  private get walletExtensionSigner(): Signer | undefined {
-    return window.coinbaseWalletExtensionSigner;
-  }
+function fetchInjectedProvider(
+  appName: string,
+  appLogoUrl: string | null
+): ProviderInterface | undefined {
+  const window = globalThis as unknown as LocalWindow;
 
-  private get coinbaseBrowser(): ProviderInterface | undefined {
-    try {
-      // Coinbase DApp browser does not inject into iframes so grab provider from top frame if it exists
-      const ethereum = window.ethereum ?? window.top?.ethereum;
-      if (!ethereum) {
-        return undefined;
-      }
+  const extension = window.coinbaseWalletExtension;
+  if (extension) {
+    if ('shouldUseSigner' in extension && extension.shouldUseSigner) return undefined;
```

Comment:
let’s make fetchInjectedProvider stateless. we don’t want any side effects (calling set***)
i know the code would look bit ugly, but passing appName and logo url seems not ideal.

how about:
```suggestion
  if (extension && !('shouldUseSigner' in extension && extension.shouldUseSigner)) {
    return extension;
```

and we might consider moving this to core/utils or somewhere there, with name something like fetchCoinbaseInjectedProvider to make it clear that it's to grab a provider injected by cb?